edward/inferences/variational_inference.py

```diff
@@ -56,22 +56,43 @@ def initialize(self, optimizer=None, var_list=None, use_prettytensor=False,
     if var_list is None:
       # Traverse random variable graphs to get default list of variables.
       var_list = set()
+      latent_var_list = set()
+      data_var_list = set()
       trainables = tf.trainable_variables()
       for z, qz in six.iteritems(self.latent_vars):
         if isinstance(z, RandomVariable):
           var_list.update(get_variables(z, collection=trainables))
+          latent_var_list.update(get_variables(z, collection=trainables))
 
         var_list.update(get_variables(qz, collection=trainables))
 
       for x, qx in six.iteritems(self.data):
         if isinstance(x, RandomVariable) and \
                 not isinstance(qx, RandomVariable):
           var_list.update(get_variables(x, collection=trainables))
+          data_var_list.update(get_variables(x, collection=trainables))
 
       var_list = list(var_list)
 
     self.loss, grads_and_vars = self.build_loss_and_gradients(var_list)
 
+    if self.logging:
+      #TODO: when var_list is not None
+      tf.summary.scalar("loss", self.loss)
+      for var in latent_var_list:
```

Comment:
Should I simply replace ':' with '_'?
The names include the scope when they are displayed on tensorboard, they don't print the full names here. 

![capture](https://cloud.githubusercontent.com/assets/10542839/24833982/c0a92db2-1ca6-11e7-8b30-97e91d6606f4.png)

But as seen from the screenshot, there is one other problem. Not all variables are grouped under scope 'model'.  It uses 'model_1' when `tf.name_scope(scope_name)` is called again.

Should I add all the variables and gradients from one scope in one go? Something like:
```
with tf.name_scope('model'):
  for grad, var in grads_and_vars:
     if var in data_var_list:
       tf.summary.histogram("parameter_" + var.name, var)
       tf.summary.histogram("gradient_" + grad.name, grad)
```
and similiarly for inference.